amlb/datasets/file.py

```diff
@@ -343,6 +343,8 @@ def __init__(self, path, fold, target, features, cache_dir, config):
         self.id_column = config['id_column']
         self.timestamp_column = config['timestamp_column']
 
+        # Ensure that id_column is parsed as string to avoid incorrect sorting
+        full_data[self.id_column] = full_data[self.id_column].astype(str)
```

Comment:
Sorry, I have been swamped lately. Hope to have a closer look soon (also the other PR).
In the current unit tests, we test for
```python
assert pat.is_categorical_dtype(ds._dtypes[ds.id_column])
```
[here](https://github.com/openml/automlbenchmark/blob/6872b63f3ebc6d7fe38ebc7376ba27c13088c78e/tests/unit/amlb/datasets/file/test_file_dataloader.py#L295C5-L295C62)

Shouldn't that test be affected?

Reply:
Thanks, that's a good catch! I've updated the test.